python/ray/serve/api.py

```diff
@@ -222,16 +222,10 @@ def deploy(self,
         else:
             raise TypeError("config must be a BackendConfig or a dictionary.")
 
-        python_methods = []
-        if inspect.isclass(backend_def):
-            for method_name, _ in inspect.getmembers(backend_def,
-                                                     inspect.isfunction):
-                python_methods.append(method_name)
-
```

Comment:
Moved this block inside `controller.deploy()` so that the autoscaler can call `controller.deploy()` without having to unpickle the backend definition from the `BackendInfo` to reconstruct the `python_methods`.

Reply:
Ah, unfortunately I don't think we can do this @architkulkarni because it means that the controller needs to load the serialized `backend_def`. We cannot do that in teh controller because the controller might run in a different env. than the backend.

Can we still do the analysis here but then store the python methods separately from the backend_def so we don't need to re-generate this metadata?

Reply:
@simon-mo actually, do we even need this python_methods thing at all? It seems like there are a lot of edge cases to it and it adds complexity. Instead, should we just proxy all `handle.<method>` calls transparently and throw an error on the called actor side?

Reply:
I would be in favor of this ^, in which case we can punt on it in this PR

Reply:
> Ah, unfortunately I don't think we can do this @architkulkarni because it means that the controller needs to load the serialized `backend_def`. We cannot do that in teh controller because the controller might run in a different env. than the backend.

Oh wow, good call. Thanks for catching that